deparse.c

```diff
@@ -1058,23 +1113,37 @@ sqlite_foreign_expr_walker(Node *node,
 			}
 			break;
 		case T_DistinctExpr:
-			/* IS DISTINCT FROM */
-			return false;
+			{
+				/* IS DISTINCT FROM */
+				elog(DEBUG2, "sqlite_fdw : %s T_DistinctExpr", __func__);
+				return false;
+			}
 		default:
+			{
 
-			/*
-			 * If it's anything else, assume it's unsafe.  This list can be
-			 * expanded later, but don't forget to add deparse support below.
-			 */
-			return false;
+				elog(DEBUG1, "sqlite_fdw : %s other", __func__);
+				/*
+				 * If it's anything else, assume it's unsafe.  This list can be
+				 * expanded later, but don't forget to add deparse support below.
+				 */
+				return false;
+			}
 	}
 
 	/*
-	 * If result type of given expression is not built-in, it can't be sent to
+	 * If result type of given expression is not built-in or PostGIS, it can't be sent to
 	 * remote because it might have incompatible semantics on remote side.
 	 */
-	if (check_type && !sqlite_is_builtin(exprType(node)))
-		return false;
+	if (check_type)
+	{
+		Oid typeOid = exprType(node);
+		if (!is_pg_builtin_Oid(typeOid))
```

Comment:
Will this better if `Oid typeOid = exprType(node);` will runs anyway, not only in case of `check_type`? `check_type=false` is rare, but `check_type=true` is also possible.

Reply:
Okay. Please keep your change. Please help me to resolve this comment.